web/client/components/TOC/fragments/settings/WMSCacheOptions.jsx

```diff
@@ -266,8 +268,41 @@ function WMSCacheOptions({
             .finally(() => setTimeout(() => setTileGridLoading(false), 500));
     };
 
+    const handleGetTileGridSettings = () => {
+        const newTileGridStrategy = layer.tileGridStrategy !== 'custom'
+            ? 'custom'
+            : undefined;
+        const promise = newTileGridStrategy === 'custom'
+            && ((layer?.tileGrids?.length || 0) === 0 || !layer?.tileGridCacheSupport)
+            ? onTileMatrixSetsFetch(layer)
+            : Promise.resolve(undefined);
+        return promise.then(({ tileGrids, tileGridCacheSupport } = {}) => {
+            const hasTileGrids = (tileGrids?.length || 0) > 0;
+            const tileGridStrategy = hasTileGrids
+                ? newTileGridStrategy
+                : undefined;
+            let tileChangedData = {
+                tileGridCacheSupport,
+                tileGridStrategy,
+                tileGrids
+            };
+            if (owner === 'background-dialog' && newTileGridStrategy === 'custom') {
+                tileChangedData = {
+                    ...tileChangedData, tiled: true, tileGridStrategy: 'custom'
+                };
+            }
+            handleOnChange(tileChangedData);
+        });
+    };
+
     const InfoText = infoText[layer.tileGridStrategy] || infoText.standard;
 
+    // fetching grid data in case background layers
+    React.useEffect(() => {
+        if (layer.remoteTileGrids && owner === 'background-dialog') {
+            handleGetTileGridSettings();
+        }
+    }, []);
```

Comment:
We should avoid to trigger a request in the component mount side effect.
Currently the catalog [epic](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/epics/catalog.js#L265-L394) that intercepts the ADD_LAYER_AND_DESCRIBE action has already all the logic that we need so I would propose to use that by making some changes to the background catalog workflow:

- Review the RecordItem [addLayer](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/catalog/RecordItem.jsx#L309-L326) method and use only the onLayerAdd function and in case of background pass all the options needed to manage the background layers in the epic
- Inside the addLayerAndDescribeEpic epic we should add the layer to the map and in case of background layer we should trigger the needed actions to display the modal (probably we will need some loading flags to alert the user when the layer is ready). This will change a bit the UX:
  - before `Click plus button catalog` -> `Modal with add button` -> `Add background layer`
  - after `Click plus button catalog`  -> `Add background layer` -> Open model for editing

Making the changes above could allow us to use the same workflow for the two catalogs.

Reply:
I totally agree. Thank you